### ISSUE TITLE
Conditional compilation symbols are not picked up in code.

### ISSUE THREAD

**Issue by melytc** [dotnet/project-system#9038]
## Visual Studio Version
Visual Studio 2022 17.7 Preview 1

## Summary
When using conditional compilation symbols, particular blocks of code in a program are compiled selectively while others are ignored.
A user has reported a [feedback ticket](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1461375) where these are not being picked up.

## Steps to Reproduce

1. Create a VB console app.
2. Open the Project Properties UI.
3. Add the following conditional compilation symbol: IsSet, True

![image](https://github.com/dotnet/project-system/assets/8518253/887ae5ee-22e9-4f29-b0c6-6b553587b7ad)

5. Go to the Program.vb file and add the following lines of code:
```vb
Imports System

Module Program
    Sub Main(args As String())
#If IsSet Then
        Console.WriteLine("Hello World!")
#End If
    End Sub
End Module
```

## Expected Behavior

No error is shown, and project can be built.

## Actual Behavior

The user gets an error saying: 
![image](https://github.com/dotnet/project-system/assets/8518253/57835c72-9de9-451d-b5fd-130ecb262e90)

As the error says, seems like the value of the key-value pair is being read as a string, instead of a boolean.
Changing the project file from:
```
<PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
    <DefineConstants>IsSet="True"</DefineConstants>
  </PropertyGroup>

  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release|AnyCPU'">
    <DefineConstants>IsSet="True"</DefineConstants>
  </PropertyGroup>
```
to:
```
<PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
    <DefineConstants>IsSet=True</DefineConstants> <-- notice no quote marks in the value
  </PropertyGroup>

  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release|AnyCPU'">
    <DefineConstants>IsSet=True</DefineConstants> <-- notice no quote marks in the value
  </PropertyGroup>
```

fixes the error and the user can build their project. However, any change in the project properties UI on the `DefineConstants` property will cause the editor to re-write the value as a string.

## User Impact
Unable to use conditional compilation in their projects through our UI when expecting boolean values.

**Comment by drewnoakes** [comment]
Does it work is the `true` value is removed from the UI, leaving only the name `IsSet`?

**Comment by melytc** [comment]
It doesn't trigger any changes to the project file, even after I change focus or edit another property:
![image](https://github.com/dotnet/project-system/assets/8518253/5a9bf0d9-b626-4b16-a9e9-490da7d01b0c)

**Comment by drewnoakes** [comment]
I think @haileymck had a fix for this recently somewhere?

**Comment by haileymck** [comment]
Closing, resolved in #8997
